packages/syntax-core/src/Typography/Typography.tsx

```diff
@@ -0,0 +1,127 @@
+import classNames from "classnames";
+import { ReactElement, ReactNode } from "react";
+import { Color } from "../constants";
+import styles from "./Typography.module.css";
+import colorStyles from "../colors/colors.module.css";
+
+function textColor(color: (typeof Color)[number]): string {
+  switch (color) {
+    case "gray700":
+      return colorStyles.gray700Color;
+    case "white":
+      return colorStyles.whiteColor;
+    case "inherit":
+      return colorStyles.inheritColor;
+    case "primary":
+      return colorStyles.primary700Color;
+    case "destructive-primary":
+      return colorStyles.destructive700Color;
+    default:
+      return colorStyles.gray900Color;
+  }
+}
+
+/**
+ * Typography is a component that renders text.
+ */
+const Typography = ({
+  align = "start",
+  as = "div",
```

Comment:
Is `as` the standard way in design systems to designate which element to use? Just curious since I hadn't seen it before

Reply:
i think it is, I've seen this around

Reply:
@jliotta depends on the design system:

* MUI uses `component`: https://mui.com/material-ui/api/box/
* NextUI uses `as`: https://nextui.org/docs/layout/container#default
* ChakraUI uses: `as`: https://chakra-ui.com/docs/components/box/usage#as-prop

So there is no consensus yet but we are also not re-inventing the wheel